demonstrations/tutorial_intro_qrom.py

```diff
@@ -0,0 +1,253 @@
+# Select operator. Below we show an example with two columns and four rows:
+#
+# .. figure:: ../_static/demonstration_assets/qrom/select_swap_4.jpeg
+#    :align: center
+#    :width: 70%
+#    :target: javascript:void(0)
+#
+# Using the same example, we have that :math:`c = |1\rangle` and :math:`r = |01\rangle`. In this case, the columns are
+# determined by a single index but we need two indexes for the rows. We invite you to check that :math:`b_5` is actually
+# loaded in the target wires.
+#
+#
+# Reusable qubits
+# ~~~~~~~~~~~~~~~~~
+#
+# The above approach has a drawback. The work wires have been altered, i.e., after applying the operator they have not
+# been returned to state :math:`|0\rangle`. This can cause unwanted behaviors, so we will present the technique shown
+# in [#cleanQROM]_ to solve this issue.
+#
+# .. figure:: ../_static/demonstration_assets/qrom/clean_version_2.jpeg
+#    :align: center
+#    :width: 90%
+#    :target: javascript:void(0)
+#
+# To see how this circuit works, let's suppose we want to load the bitstring :math:`b_{cr}` in the target wires, where :math:`cr` is the integer whose binary representation is :math:`|c\rangle|r\rangle`.
+# We can summarize the idea in a few simple steps:
+#
+# 1. **We start by generating the uniform superposition on the r-th register of the work wires**. To do this, we put the Hadamard in the target wires and move it to the :math:`r` -row with the swap block. We denote by :math:`R` the number of rows.
+#
+# .. math::
+#       |c\rangle |r\rangle |0\rangle_0 |0\rangle_1 \dots |+\rangle_r \dots |0\rangle_{R-1}
+#
+# 2. **We apply the Select block.** Note that in the :math:`r`-th position, the Select has no effect since the state :math:`|+\rangle` is not modified by :math:`X` gates.
+#
+# .. math::
+#       |c\rangle |r\rangle |b_{c0}\rangle_0 |b_{c1}\rangle_1 \dots |+\rangle_r \dots |b_{c(R-1)}\rangle_{R-1}
+#
+#
+# 3. **We apply the Hadamard's in r-th register of the work wires.** The two swap blocks and the Hadamard gate in target wires achieve this.
+#
+# .. math::
+#       |c\rangle |r\rangle |b_{c0}\rangle_0 |b_{c1}\rangle_1 \dots |0\rangle_r \dots |b_{c(R-1)}\rangle_{R-1}
+#
+# 4. **We apply Select again to the state.** Note that loading the bitstring twice in the same register leaves the state as :math:`|0\rangle`. (:math:`X^2 = \mathbb{I}`)
+#
+# .. math::
+#       |c\rangle |r\rangle |0\rangle_0 |0\rangle_1 \dots |b_{cr}\rangle_r \dots |0\rangle_{R-1}
+#
+# That's it! With a last swap we have managed to load the bitstring of column :math:`c` and row :math:`r` in the target wires.
+#
+# QROM in PennyLane
+# -----------------
+# Coding a QROM circuit from scratch can be painful, but with the help of PennyLane you can do it in just one line of code.
+# Let's see an example where we encode longer bitstrings and we will use enough work wires to group four bitstrings per column:
+
+bitstrings = ["01", "11", "11", "00", "01", "11", "11", "00"]
+
+control_wires = [0, 1, 2]
```

Comment:
To make a parallel with the explanation in the text, one can also divide the control wires as:

```suggestion
column_wires = [0,1]
row_wires = [2]
control_wires = column_wires + row_wires
```

Reply:
Good idea! However I'll leave it as it is because I hope the demo will help people to know how to use the template (and that would not be the expected use)